app/src/main/java/edu/uc/kovaciad/slicetracker/dao/IBrandDAO.kt

```diff
@@ -17,8 +15,21 @@ interface IBrandDAO {
 
     @Insert
     fun insertBrand(brand: Brand)
+    /**
+     * When updating a row with a value already set in a column,
+     * replaces the old value with the new one.
+     *
+     * @param brand new value to write
+     */
+    @Update
+    suspend fun update(brand: Brand)
 
-    @Delete
-    fun deleteBrand(brand: Brand)
+    /**
+     * Deletes all values from the table.
+     *
+     * This does not delete the table, only its contents.
+     */
+    @Query("DELETE FROM Brand")
+    suspend fun clear()
```

Comment:
For IBrandDAO and IMaterialDAO I added a update function because when updating a row with a value already set in a column this replaces the old value with the new one. Also, I modified your delete function. It deletes all values from the table. This does not delete the table, only its contents.
Check out: https://developer.android.com/reference/androidx/room/Update
https://developer.android.com/reference/kotlin/androidx/room/Delete